File: Diary.DAL/Models/Message.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Diary.DAL.Models
{
    public class Message
    {
        public long Id { get; set; }
        public string Text { get; set; }
        public bool IsRead { get; set; }
        public DateTime TimeSent { get; set; }
        [ForeignKey("SenderUser")]
        public long? SenderUserId { get; set; }
        [ForeignKey("ReceiverUser")]
        public long? ReceiverUserId { get; set; }
        public User SenderUser { get; set; }
        public User ReceiverUser { get; set; }

    }
}

File: ElectronicDiary/app/components/message/Message.ts
export class Message {
    Id: number;
    Text: string;
    IsRead:boolean;
    TimeSent: Date;
    SenderUserId: number;
    SenderUserName:string;
    ReceiverUserId: number;
    ReceiverUserName: string;
    IsSent:boolean;
}